ui/src/components/VoxettaRecordButton.js

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+import {LitElement, html, css} from 'lit-element';
+
+import {AudioRecorder} from '../utils/AudioRecorder';
+import {UtteranceApiService} from '../utils/UtteranceApiService';
+
+import {Icon} from '@material/mwc-icon';
+
+/**
+ * Button responsible for enabling the user to record and upload audio files. 
+ */
+export class VoxettaRecordButton extends LitElement {
+    static get properties() {
+        return {
+            isRecording: {type: Boolean},
+            audioStream: {type: Object}
+        };
+    }
+
+    static get styles() {
+        return css`
+            div {
+                align-items: center; 
+                display: flex; 
+                flex-direction: column; 
+                flex-wrap: wrap;
+                justify-content: center; 
+                text-align: center;  
+            }
+            mwc-icon-button {
+                --mdc-icon-button-size: 80px;
+                --mdc-icon-size: 40px; 
+                border-radius: 50%;
+                box-shadow: 0px 8px 15px #dcdcdc;
+                color: #3c4043;
+            }
+        `;
+    }
+
+    constructor() {
+        super();
+        this.isRecording = false;
+        this.audioRecorder = new AudioRecorder();
+        this.utteranceService = new UtteranceApiService();
+        this.audioStream; 
+    }
+  
+    /**
+     * If the user is not currently recording, begin recording using the Microphone 
+     * component. Otherwise, stop recording and save and display the just-recorded 
+     * audio file.
+     */
+    async recordHandler() {
+        if (!this.isRecording) {
+            try { 
+                await this.audioRecorder.initRecorder();
+            } catch(e) { 
+                console.log('Microphone blocked.'); 
+                return;
+            }
+
+            if (!this.audioRecorder.startRecording()) {
+                 console.log('Failed to start recording.')
+                 return; 
+            }
+            this.isRecording = true;
+            this.styleOnButton();
```

Comment:
Instead of setting the style programmatically, use styleMap to update styles.
https://lit-html.polymer-project.org/guide/styling-templates
You can create the styles with a function though
```
<button style=${styleMap(recordingStyles)}>
```

Reply:
changed to use styleMap